python/cugraph/cugraph/sampling/uniform_neighbor_sample.py

```diff
@@ -17,10 +17,17 @@
 import numpy
 
 import cudf
+import cupy as cp
 
 
 def uniform_neighbor_sample(
-    G, start_list, fanout_vals, with_replacement=True, is_edge_ids=False
+    G,
+    start_list,
+    fanout_vals,
+    with_replacement=True,
+    with_edge_properties=False,
+    batch_id_list=None,
+    seed=42,
```

Comment:
Not sure if providing a default seed is a good idea as this is written.  If the approach suggested above were to be used (creating some sort of stateful randomization object), having a default seed for the stateful object might be reasonable.  But providing a default here will not provide good random samples except in the very first call.

Reply:
So is the way to rectify this requiring that the user provide a seed?  Or do we instead associate a random state with a graph object?  Or do we use a random default seed?

Reply:
I don't know what the right solution is from a user experience perspective.  My thought would be:
* Create a stateful python object that represents random number generator state.
* The constructor for that stateful python object would take a seed as a parameter, perhaps defaulting to some value if not provided
* The stateful python object would instantiate a `cugraph_rng_state_t`, initializing it with the seed.  In MG mode, the state would be initialized with `seed + rank`
* The stateful python object would be used by `uniform_neighbor_sample` to send the `cugraph_rng_state_t` into the C API.  Note that the `cugraph_rng_state_t` is modified by the C API, it is updated each time we use it to generate a random number to a new state to create a sequence of random numbers.  Reusing it keeps the sequence of random numbers flowing properly.

I'm not terribly familiar with python semantics.  This stateful python object could be managed by the user (required as a parameter and visible in the python API).  I suppose you could also make it a singleton and hide it from the API, just providing some public function to initialize the seed once.

Reply:
@ChuckHastings I exposed the random state at the PLC level, and added a global random state.  In `pylibcugraph.uniform_neighbor_sample`, the user can provide the random state, or it defaults to the global state, instantiating it if necessary.  Please confirm this is an acceptable solution.

Reply:
Sorry I'm late to this conversation, but the best UX in my opinion would be to be consistent with RAPIDS and accept a numpy or cupy `RandomState` instance, as both [cuDF](https://docs.rapids.ai/api/cudf/stable/api_docs/api/cudf.series.sample) and [cuML](https://docs.rapids.ai/api/cuml/stable/api.html#model-selection-and-data-splitting) do.

Reply:
Wouldn't that have to be changed at the C++ level?

Reply:
Possibly, I haven't looked into their implementations.  One option would be to rename the `rng_state` param `random_state` like the other libraries call it and accept an int (also like they do). Then eventually also accept a cupy/numpy `RandomState` instance at the python layers. This will give us a path to fully supporting it, consistent with the other libs, without breaking backwards compat.

For now, PLC could still pass a `CuGraphRandomState` that it creates with the int passed in, or the default global instance if not passed in.  It might be nice to avoid exposing `CuGraphRandomState` in the python API anyway since it's apparently non-standard.

Reply:
@rlratzel @ChuckHastings based on our libs team discussion this morning, I am going to revert to accepting a seed argument, and not use a global random state.  At some point, we can support a cupy/numpy `RandomState` and properly transmute it at the PLC layer.

Reply:
cc @jnke2016